lib/SILOptimizer/LoopTransforms/ArrayPropertyOpt.cpp

```diff
@@ -356,12 +360,10 @@ class ArrayPropertiesAnalysis {
     return false;
   }
 
-  SmallPtrSetImpl<SILBasicBlock *> &getLoopExitingBlocks() {
+  SmallVectorImpl<SILBasicBlock *> &getLoopExitingBlocks() {
     if (!CachedExitingBlocks.empty())
       return CachedExitingBlocks;
-    SmallVector<SILBasicBlock *, 16> ExitingBlocks;
-    Loop->getExitingBlocks(ExitingBlocks);
-    CachedExitingBlocks.insert(ExitingBlocks.begin(), ExitingBlocks.end());
+    Loop->getExitingBlocks(CachedExitingBlocks);
```

Comment:
Question. Do you know why we used a SmallPtrSet here? Also, I imagine that this can be returned as an ArrayRef.

Reply:
I haven't thought about it, I am just double checking, I don't think our loop can have multiple of the same blocks passed here, right? Maybe we need an assert to that effect? Your thoughts?

Reply:
>  Do you know why we used a SmallPtrSet here?

I think this was just a bug.

>  I imagine that this can be returned as an ArrayRef.

true. I'll change it.

> I don't think our loop can have multiple of the same blocks passed here, right?

yes

> Maybe we need an assert to that effect? Your thoughts?

I think an assert is not needed (it would be not trivial).

Reply:
I actually think the assert wouldn't be too difficult (consider a copy + sortUnique + ensure nothing removed), that being said I think such an assert would be something for a lower part of the API.